### PR TITLE
IDAM-307 API endpoints for roles

### PR DESCRIPTION
**Pull request by josealmeida**
https://nicedigital.atlassian.net/browse/IDAM-307

### CODE REVIEW

File: NICE.Identity.Authorisation.WebAPI/Controllers/UserRolesController.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using NICE.Identity.Authorisation.WebAPI.ApiModels;
+using NICE.Identity.Authorisation.WebAPI.Services;
+
+namespace NICE.Identity.Authorisation.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
```
Comment (John-Williams) [inline NICE.Identity.Authorisation.WebAPI/Controllers/UserRolesController.cs]: i think this needs a policy on it.

File: NICE.Identity.Authorisation.WebAPI/Controllers/RolesController.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using NICE.Identity.Authorisation.WebAPI.ApiModels;
+using NICE.Identity.Authorisation.WebAPI.Services;
+
+namespace NICE.Identity.Authorisation.WebAPI.Controllers
+{
+    [Route("api/[controller]")]
+    [Authorize(AuthenticationSchemes = JwtBearerDefaults.AuthenticationScheme)]
```
Comment (John-Williams) [inline NICE.Identity.Authorisation.WebAPI/Controllers/RolesController.cs]: missing policy here i think

File: NICE.Identity.Authorisation.WebAPI/Controllers/UserRolesController.cs (hunk 1 above)
Comment (josealmeida) [reply NICE.Identity.Authorisation.WebAPI/Controllers/UserRolesController.cs]: Changes done

File: NICE.Identity.Authorisation.WebAPI/Controllers/RolesController.cs (hunk 2 above)
Comment (josealmeida) [reply NICE.Identity.Authorisation.WebAPI/Controllers/RolesController.cs]: Changes done.

**Pull request merged**